src/workerd/api/node/crypto_random-test.js

```diff
@@ -0,0 +1,364 @@
+  const buf = Buffer.from(prime);
+  const val = buf.readUInt32BE();
+  strictEqual(val % add, rem);
+}
+
+{
+  const prime = generatePrimeSync(32, { add: BigInt(add), rem: BigInt(rem) });
+  ok(checkPrimeSync(prime));
+  const buf = Buffer.from(prime);
+  const val = buf.readUInt32BE();
+  strictEqual(val % add, rem);
+}
+
+/*{
+  // The behavior when specifying only add without rem should depend on the
+  // safe option.
+
+  // Do not seem to work with boringssl
+  if (process.versions.openssl >= '1.1.1f') {
+    generatePrime(128, {
+      bigint: true,
+      add: 5n
+    }, mustSucceed((prime) => {
+      ok(checkPrimeSync(prime));
+      strictEqual(prime % 5n, 1n);
+    }));
+
+    generatePrime(128, {
+      bigint: true,
+      safe: true,
+      add: 5n
+    }, mustSucceed((prime) => {
+      ok(checkPrimeSync(prime));
+      strictEqual(prime % 5n, 3n);
+    }));
+  }
+}*/
+
+{
+  // This is impossible because it implies (prime % 2**64) == 1 and
+  // prime < 2**64, meaning prime = 1, but 1 is not prime.
+  for (const add of [2n ** 64n, 2n ** 65n]) {
+    throws(() => {
+      generatePrimeSync(64, { add });
+    }, {
+      name: 'RangeError'
+    });
+  }
+
+  // Any parameters with rem >= add lead to an impossible condition.
+  for (const rem of [7n, 8n, 3000n]) {
+    throws(() => {
+      generatePrimeSync(64, { add: 7n, rem });
+    }, {
+      name: 'RangeError'
+    });
+  }
+
+  // This is possible, but not allowed. It implies prime == 7, which means that
+  // we did not actually generate a random prime.
+  throws(() => {
+    generatePrimeSync(3, { add: 8n, rem: 7n });
+  }, {
+    name: 'RangeError'
+  });
+
+  /*if (process.versions.openssl >= '1.1.1f') {
+    // This is possible and allowed (but makes little sense).
+    strictEqual(generatePrimeSync(4, {
+      add: 15n,
+      rem: 13n,
+      bigint: true
+    }), 13n);
+  }*/
+}
+
+[1, 'hello', {}, []].forEach((i) => {
+  throws(() => checkPrime(i), {
+    code: 'ERR_INVALID_ARG_TYPE'
+  });
+});
+
+for (const checks of ['hello', {}, []]) {
+  /*throws(() => checkPrime(2n, { checks }, mustNotCall()), {
+    code: 'ERR_INVALID_ARG_TYPE',
+    message: /checks/
+  });*/
+  throws(() => checkPrimeSync(2n, { checks }), {
+    code: 'ERR_INVALID_ARG_TYPE',
+    message: /checks/
+  });
+}
+
+for (const checks of [-(2 ** 31), -1, 2 ** 31, 2 ** 32 - 1, 2 ** 32, 2 ** 50]) {
+  /*throws(() => checkPrime(2n, { checks }, mustNotCall()), {
+    code: 'ERR_OUT_OF_RANGE',
+    message: /<= 2147483647/
+  });*/
+  throws(() => checkPrimeSync(2n, { checks }), {
+    code: 'ERR_OUT_OF_RANGE',
+    message: /<= 2147483647/
+  });
+}
+
+ok(
+  !checkPrimeSync(
+    Buffer.from([0x1]),
+    {
+      fast: true,
+      trialDivision: true,
+      checks: 10
+    }));
+
+throws(() => {
+  generatePrimeSync(32, { bigint: '' });
+}, { code: 'ERR_INVALID_ARG_TYPE' });
+
+/*throws(() => {
+  generatePrime(32, { bigint: '' }, mustNotCall());
+}, { code: 'ERR_INVALID_ARG_TYPE' });*/
```

Comment:
For the `mustNotCall()` pattern, something like...

```
const p = deferredPromise(); // see the the streams test for deferredPromise impl
throws(() => {
  generatePrime(32, { bigint: '' }, p.reject);
}, { code: 'ERR_INVALID_ARG_TYPE' });
Promise.resolve().then(() => p.resolve());
await p;
```

It might look at bit weird but since a promise can only be resolved once, and we are deferring the callback to the microtask queue, this should structure things such that if the callback is called, p.reject will happen before the p.resolve and the await p will throw.

Reply:
Or, just pass a non-op function as the callback and ignore whether or not the callback is invoked.